card/mod/03_machines/lib/card/machine.rb

```diff
@@ -74,7 +74,7 @@ def self.define_machine_events host_class
       event_suffix = host_class.name.tr ':', '_'
       event_name = "reset_machine_output_#{event_suffix}".to_sym
       host_class.event event_name, after: :expire_related, on: :save do
-        reset_machine_output!
+        reset_machine_output
```

Comment:
yes, and thousand other things 

I removed the ! in the method name. I think the name makes clear what it does and there is nothing dangerous or unexpected.